classes/local/apibridge.php

```diff
@@ -1791,4 +1849,226 @@ public function get_duplicated_episodeid($workflowid) {
+                        $acl[] = $roleobject;
+                    }
+                }
+            } else {
+                foreach ($role->actions as $action) {
+                    // Define role object.
+                    $roleobject = (object)array('allow' => true,
+                        'role' => $this->replace_placeholders($role->rolename, $courseid)[0],
+                        'action' => $action);
+                    
+                    // Check if the role object already exists in the acl list.
+                    $existingacl = array_filter($acl, function($v, $k) use ($roleobject) {
+                        if ($v->role == $roleobject->role && $v->action == $roleobject->action) {
+                            return true;
+                        }
+                    },ARRAY_FILTER_USE_BOTH);
+
+                    // In case the role object is new, we add it to the acl list. This helps making a clean list.
+                    if (empty($existingacl)) {
+                        $acl[] = $roleobject;
+                    }
+                }
+            }
+        }
+
+        // Put everything in params value as a string.
+        $params['acl'] = json_encode(array_values($acl));
+
+        // When there is nothing to change.
+        if ($params['acl'] == ($jsonacl)) {
+            return true;
+        }
+
+        // Update the acls via put request.
+        $api = new api();
+        $api->oc_put($resource, $params);
+
+        // Finally we return the result of that request to the server.
+        return ($api->get_http_code() == 200);
+    }
+
+    /**
+     * Make Event visible for the new course that has been imported to, by changing the ACLs.
+     *
+     * @param string $identifier event identifier
+     * @return int $courseid id of the course being imported to.
+     */
+    private function imported_events_acl_change($identifier, $courseid) {
+        // Use try to catch unwanted errors.
+        try {
+            // Make it visible to the course does the ACL change accordingly.
+            $visibilychanged = $this->change_visibility($identifier, $courseid, block_opencast_renderer::VISIBLE);
+            // In order to resolve the return result of the change_visibilty method, we assume non (false) values as true.
+            return ($visibilychanged !== false) ? true : false;
+        } catch (\moodle_exception $e) {
+            // It is unlikely, but who knows.
+            return false;
+        }
+    }
+
+    /**
+     * Map the seriesid to the course that has been imported to, by assinging the series as secondary.
+     *
+     * @return int $courseid id of the course being imported to.
+     * @param string $seriesid series id.
+     */
+    private function map_imported_series_to_course($courseid, $seriesid) {
+        try {
+            // Get the current record.
+            $mapping = seriesmapping::get_record(array('courseid' => $courseid, 'series' => $seriesid, 'isdefault' => '0'));
```

Comment:
It would be better if you're using `seriesmapping::get_record(array('courseid' => $courseid, 'series' => $seriesid), true)`. This allows you to find any mapping for the course/series independent of the default status. That's important as we don't want to have the same series twice in a course.

The `true` must be added because things are a bit complicated with the multiple series features because I try to guarantee backwards compatibility. The `true` parameters allows you to query the database without setting a filter on `isdefault`.